logstash-core/lib/logstash/java_pipeline.rb

```diff
@@ -255,6 +259,19 @@ def start_workers
     end
   end
 
+  def resolve_cluster_uuids
+    cluster_uuids = []
+    outputs.each do |output|
+      if LogStash::SETTINGS.registered?(output.id + ".cluster_uuid")
```

Comment:
Would you mind pointing me to the place where these settings are dynamically registered? I'm not finding `cluster_uuid` anywhere on current `master`.

IIRC, `LogStash::SETTINGS#validate_all` will cause a failure when a setting is provided in the yaml or at command-line and the relevant setting has not been registered.

Reply:
@yaauie Certainly. This requires a little bit of background knowledge on a discussion that @jsvd had. The setting is registered by the Elasticsearch plugin. The PR which introduces this functionality is here: https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/857

That said, it's totally possible that I don't fully grok how settings _should_ be registered so if this is not the correct manner for doing so, I'd appreciate any advice. Thanks!

Reply:
As discussed in other threads, I consider routing through `LogStash::SETTINGS` to stash plugin metadata to be a hack, and while it can be made to work I'd much rather provide you with the tools you need to support this feature in a way that is less fragile.

 - PR introducing `Plugin#plugin_metadata` and the metadata registry is https://github.com/elastic/logstash/pull/10636
 - Notes about implementing on top of my PR are [on your other PR](https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/857#issuecomment-479706585)